### PR TITLE
APEXCORE-726 Setup up acls for impersonation

### PR DESCRIPTION
**Pull request by pramodin**
@devtagare @amberarrow please see

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/security/ACLManager.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (tweise) [inline engine/src/main/java/com/datatorrent/stram/security/ACLManager.java]: new code needs to go into org.apache.apex packages

**Review by tweise — CHANGES_REQUESTED** [review]

File: engine/src/main/java/com/datatorrent/stram/StramClient.java [hunk 2]
```diff
@@ -37,7 +37,6 @@
 import org.slf4j.LoggerFactory;
 
 import org.apache.apex.common.util.JarHelper;
-
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/StramClient.java]: Avoid format changes

File: engine/src/main/java/com/datatorrent/stram/security/ACLManager.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    if (areAclsRequired(conf)) {
+      logger.debug("Configuring ACLs for {}", userName);
+      Map<ApplicationAccessType, String> acls = Maps.newHashMap();
+      acls.put(ApplicationAccessType.VIEW_APP, userName);
+      acls.put(ApplicationAccessType.MODIFY_APP, userName);
+      launchContext.setApplicationACLs(acls);
+    }
+  }
+
+  public static boolean areAclsRequired(Configuration conf)
```
Comment (sanjaypujare) [inline engine/src/main/java/com/datatorrent/stram/security/ACLManager.java]: Should this be named `areACLsRequired` ? We should be consistent in ACL vs Acl in identifier names

**Comment by pramodin** [comment]
squashed

**Review by devtagare — APPROVED** [review]
LGTM

**Pull request merged**